charts/k8s-monitoring/charts/feature-pod-logs/templates/_filelog.alloy.tpl

```diff
@@ -0,0 +1,154 @@
+      from      = "pod"
+    }
+{{- range $attribute, $annotation := .Values.annotations }}
+    annotation {
+      tag_name = {{ $attribute | quote }}
+      key      = {{ $annotation | quote }}
+      from     = "pod"
+    }
+{{- end }}
+{{- range $attribute, $annotation := .Values.nodeAnnotations }}
+    annotation {
+      tag_name = {{ $attribute | quote }}
+      key      = {{ $annotation | quote }}
+      from     = "node"
+    }
+{{- end }}
+{{- range $attribute, $annotation := .Values.namespaceAnnotations }}
+    annotation {
+      tag_name = {{ $attribute | quote }}
+      key      = {{ $annotation | quote }}
+      from     = "namespace"
+    }
+{{- end }}
+    label {
+      key_regex = "(.*)"
+      tag_name  = "$1"
+      from      = "pod"
+    }
+{{- range $attribute, $label := .Values.labels }}
+    label {
+      tag_name = {{ $attribute | quote }}
+      key      = {{ $label | quote }}
+      from     = "pod"
+    }
+{{- end }}
+{{- range $attribute, $label := .Values.nodeLabels }}
+    label {
+      tag_name = {{ $attribute | quote }}
+      key      = {{ $label | quote }}
+      from     = "node"
+    }
+{{- end }}
+{{- range $attribute, $label := .Values.namespaceLabels }}
+    label {
+      tag_name = {{ $attribute | quote }}
+      key      = {{ $label | quote }}
+      from     = "namespace"
+    }
+{{- end }}
+  }
+
+  output {
+    logs = [otelcol.processor.transform.pod_logs.input]
+  }
+}
+
+otelcol.processor.transform "pod_logs" {
+  error_mode = "ignore"
+  log_statements {
+    context = "resource"
+    statements = [
+      `delete_key(attributes, "k8s.container.restart_count")`,
+      `delete_key(attributes, "log.file.path")`,
+
+      `set(attributes["service.name"], attributes["service_name"]) where attributes["service.name"] == nil`,
+      `set(attributes["service.name"], attributes["service"]) where attributes["service.name"] == nil`,
+      `set(attributes["service.name"], attributes["app"]) where attributes["service.name"] == nil`,
+      `set(attributes["service.name"], attributes["application"]) where attributes["service.name"] == nil`,
+      `set(attributes["service.name"], attributes["name"]) where attributes["service.name"] == nil`,
+      `set(attributes["service.name"], attributes["app.kubernetes.io/name"]) where attributes["service.name"] == nil`,
+      `set(attributes["service.name"], attributes["container"]) where attributes["service.name"] == nil`,
+      `set(attributes["service.name"], attributes["container_name"]) where attributes["service.name"] == nil`,
+      `set(attributes["service.name"], attributes["k8s.container.name"]) where attributes["service.name"] == nil`,
+      `set(attributes["service.name"], attributes["component"]) where attributes["service.name"] == nil`,
+      `set(attributes["service.name"], attributes["workload"]) where attributes["service.name"] == nil`,
+      `set(attributes["service.name"], attributes["job"]) where attributes["service.name"] == nil`,
```

Comment:
ideally we try to match what the opentelemetry operator is doing here:
https://github.com/open-telemetry/opentelemetry-operator?tab=readme-ov-file#how-servicename-is-calculated